### PR TITLE
Provide simpler route for activating shell integration

### LINKED ISSUE #153921: Simplify shell integration manual install by deferring to the code CLI
Currently install is a little cumbersome, for example:

> Add the following to your [PowerShell profile](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.2), replacing `<InstallDir>` with VS Code's installation directory. Run `code $Profile` in pwsh to open the file in VS Code.
>```pwsh
>if ($env:TERM_PROGRAM -eq "vscode") {
>  . "<InstallDir>\resources\app\out\vs\workbench\contrib\terminal\browser\media\shellIntegration.ps1"
>}
>```

It would be much nicer if this was encapsulated within the CLI, something like:

```
code shell-integration
```

We should also make sure these cases are covered:

- Remote development extensions - injection works currently but manual install doesn't talk about the server paths
- Regular SSH - it's not so easy to get the shell integration script to the remote machine

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #153921

```pwsh
# pwsh
if ($env:TERM_PROGRAM -eq "vscode") { . "$(code --shell-integration pwsh)" }
```

```sh
# bash
[[ "$TERM_PROGRAM" == "vscode" ]] && . "$(code --shell-integration bash)"

# zsh
[[ "$TERM_PROGRAM" == "vscode" ]] && . "$(code --shell-integration zsh)"
```

### CODE REVIEW

File: scripts/code-cli.bat [hunk 1]
```diff
@@ -6,7 +6,7 @@ title VSCode Dev
 pushd %~dp0..
 
 :: Get electron, compile, built-in extensions
-if "%VSCODE_SKIP_PRELAUNCH%"=="" node build/lib/preLaunch.js
+@REM if "%VSCODE_SKIP_PRELAUNCH%"=="" node build/lib/preLaunch.js
```
Comment (Tyriar) [inline scripts/code-cli.bat:9]: Need to remove changes to this file

**Comment by felipecrs** [comment]
Another approach would be to output the whole script when calling such command instead of script path. Like Homebrew's shellenv:

```zsh
eval "$(brew shellenv)"
```

Or other tools which generates completion scripts, like:

```
kubectl completion zsh
```

**Comment by Tyriar** [comment]
@felipecrs I think since we'll say to use the if term_program is vscode, the eval should work the same as sourcing the file

As for completions, shell integration doesn't need completions to work, but we have some early prototypes for how to tackle that problem starting with pwsh https://github.com/microsoft/vscode/issues/154662

**Review by meganrogge — APPROVED** [review]
Did not test but lgtm

**Pull request merged**